compiler/compile/OMRSymbolReferenceTable.cpp

```diff
@@ -989,30 +989,46 @@ OMR::SymbolReferenceTable::findOrCreateSymRefWithKnownObject(TR::SymbolReference
    }
 
 TR::SymbolReference *
-OMR::SymbolReferenceTable::findOrCreateSymRefWithKnownObject(TR::SymbolReference *original, TR::KnownObjectTable::Index objectIndex)
+OMR::SymbolReferenceTable::findTempSymRefWithKnownObject(TR::KnownObjectTable::Index knownObjectIndex)
    {
-   TR_BitVector *bucket = _knownObjectSymrefsByObjectIndex[objectIndex];
+   return findSymRefWithKnownObject(NULL, knownObjectIndex, true /* isTemp */);
+   }
+
+TR::SymbolReference *
+OMR::SymbolReferenceTable::findSymRefWithKnownObject(TR::Symbol *symbol, TR::KnownObjectTable::Index knownObjectIndex, bool isTemp)
+   {
+   TR_BitVector *bucket = _knownObjectSymrefsByObjectIndex[knownObjectIndex];
```

Comment:
why does find allocate the bucket - if we are finding and there is no bucket we should just return NULL right?

Reply:
can't we tell from the symbol if we are dealing with a temp rather than having a Boolean flag?

Reply:
Actually the symbol would be NULL if it's a temp. I thought passing a flag would be more straight forward than just checking if the symbol is NULL it must be a temp.

Reply:
I'm still not sure I like the bool and the symbol needing to agree - it seems redundant....

Reply:
yea, I agree it does seem redundant... changed to check if the symbol is NULL and added a comment to avoid confusion.